bagel/dataset_description_model.py

```diff
@@ -82,12 +91,46 @@ class DatasetDescription(BaseModel):
         ),
     ]
     access_link: Annotated[
-        AnyHttpUrl,
+        HttpUrl,
         Field(
             default=None,
             description="Primary link for access requests or information.",
             alias="AccessLink",
         ),
     ]
 
-    model_config = ConfigDict(extra="ignore")
+    # NOTE: url_preserve_empty_path (>=2.12) is needed to prevent HttpUrl from auto-appending trailing slashes to URLs
+    # when the path portion of the URL is empty
+    # (https://docs.pydantic.dev/latest/migration/#url-and-dsn-types-in-pydanticnetworks-no-longer-inherit-from-str)
+    model_config = ConfigDict(extra="ignore", url_preserve_empty_path=True)
+
+    @field_validator("name")
+    @classmethod
+    def check_name_not_whitespace(cls, value: str) -> str:
+        """
+        Raise an error (will be caught as a ValidationError by Pydantic) if
+        the required 'Name' field is an empty string or all whitespace.
+        """
+        if value.strip() == "":
+            raise ValueError("'Name' field cannot be an empty string.")
+        return value
+
+    @field_validator("access_instructions")
+    @classmethod
+    def whitespace_string_to_default_none(
+        cls, value: str | None
+    ) -> str | None:
+        """Convert an empty string or a string that contains only whitespace to None."""
+        if value is not None and value.strip() == "":
+            return None
+        return value
+
```

Comment:
so iiuc, this is one way for you to enable a fallback default value if the user provides a nonsensical input (e.g. empty string)? I'm wondering: what will we do with this in practice? If the user added "access_instructions" as empty string, and we don't warn them about it / error out, what are we going to do with the info?

Reply:
Thanks @alyssadai. I think that is definitely enough then to let the user know they should probably fix the file!